src/rustdoc_schema.graphql

```diff
@@ -694,6 +694,37 @@ type Constant implements Item & Importable & GlobalValue {
   docs: String
   attrs: [String!]!
   visibility_limit: String!
+  # properties for Constant
+  """
+  A type represented in the "raw" rustdoc JSON representation, see interface RawType.
+  Currently serialized as a serde_json String instead of a proper RawType
+  ```rust
+  //          	                	            	// rustdocs type field
+  const MIN   	:          usize	= 16       ;	// {"primitive":"usize"}
+  const LOG_AS	: &'static str  	= "batch"  ;	// {"borrowed_ref":{"lifetime":"'static","mutable":false,"type":{"primitive":"str"}}}
+  const YEAR  	:          Years	= Years(42);	// {"resolved_path":{"name":"Years","id":"0:3:1633","args":{"angle_bracketed":{"args":[],"bindings":[]}}}}
+  ```
+  """
+  type_: String
```

Comment:
I'd advise against this, and in favor of an alternative approach I think is more elegant.

Instead of needing code to unpack this kind of serde-in-serde representation, consider having your use case grab the `id` value here and use it to look up the item in the [`index` of the deserialized rustdoc value](https://docs.rs/rustdoc-types/latest/rustdoc_types/struct.Crate.html#structfield.index) — the same one backing the queries here.

That way has a few advantages:
- You'll get a type-safe way to interact with the complex type information of constants.
- The code won't need to sneak serialized JSON into a string field, which as you've noted causes ergonomics problems.
- Once Trustfall supports custom scalar types, it gives us the option of adding a "proper" `type_` field to the schema that will be more ergonomic. (No concrete timeline on this, though I'd love to direct my time toward building it if some funding were available.)

Reply:
ok, will check the id way, meanwhile removed type from Assoc/Const properties